eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventlist/EventListActivity.java

```diff
@@ -80,20 +79,20 @@ protected void onCreate(Bundle savedInstanceState)
         mScreenView = new EvenListScreenViewImpl(this);
         mScreenView.init();
 
-        RetainedObjects retainedObjects = new Restoring(getLastCustomNonConfigurationInstance());
-
         mApiService = new FutureLocalServiceConnection<>(this,
                 new Intent("com.schedjoules.API").setPackage(getPackageName()));
 
-        mEventListItems = new EventListItemsImpl();
-        mListItemsProvider = retainedObjects.getOr(0,
-                new EventListItemsProviderImpl(mApiService, mEventListItems,
-                        new EventListBackgroundMessage(this), new EventListLoadingIndicatorOverlay(this)));
+        RetainedObjects retainedObjects = new Restoring(getLastCustomNonConfigurationInstance());
+        mListItemsProvider = retainedObjects.getOr(0, new EventListItemsProviderImpl(new EventListItemsImpl()));
+
+        mListItemsProvider.setApiService(mApiService);
```

Comment:
Just some more thoughts on this:

After looking at current solution again and comparing to retained Fragment, I think now that that would be better, so if we need to touch this part again, I think it might be worth moving to Fragment. The reason is that in the Fragment all the objects that are needed to be retained would be initialized in onCreate(), and the view and other context specific ones in onCreateView(). Basically no extra magic needed as now.

About what needs to be retained now - isn't it the event list the main reason why don't want to do serialization now? I am not sure but wouldn't Serializing (/parcelling) possibly hundreds of events take noticeable time?
And currently all the event list loading mechanism and all the related state is deliberately contained in this `EventListItemProviderImpl`. So if we have to retain any of that state, it looks practical to just retain the whole object.

I checked out the `AuthorizationCodeGrant` state handling, I see the point there, looks like an interesting pattern, so I will keep it in mind.